pkg/storage/utils/decomposedfs/lookup/lookup.go

```diff
@@ -44,6 +44,20 @@ func init() {
 	tracer = otel.Tracer("github.com/cs3org/reva/pkg/storage/utils/decomposedfs/lookup")
 }
 
+// PathLookup defines the interface for the lookup component
+type PathLookup interface {
+	NodeFromResource(ctx context.Context, ref *provider.Reference) (*node.Node, error)
+	NodeFromID(ctx context.Context, id *provider.ResourceId) (n *node.Node, err error)
+
+	InternalRoot() string
+	InternalPath(spaceID, nodeID string) string
+	Path(ctx context.Context, n *node.Node, hasPermission node.PermissionFunc) (path string, err error)
+	MetadataBackend() metadata.Backend
+	ReadBlobSizeAttr(ctx context.Context, path string) (int64, error)
+	ReadBlobIDAttr(ctx context.Context, path string) (string, error)
+	TypeFromPath(ctx context.Context, path string) provider.ResourceType
+}
+
```

Comment:
Can we get rid of the interface entirely? I don't see a reason for keeping it - Reimplementing such a big interface will not be done anyways...

Reply:
Maybe. But it's not only about enabling reimplementations but also about decoupling dependencies (just removing it currently leads to import cycles) and increasing testability (albeit I'm not sure whether the latter is the case here.
We can take a look at that, the lookup interface has been a nuisance a few times already, but I'd rather we do that as a separate change as it's gonna be a bigger refactoring.